### DISCUSSION TITLE
Can I extract current user from within a custom plugin?

### DISCUSSION THREAD

**Discussion by lomshabhishek** [DigitalSlideArchive/digital_slide_archive#297] [Q&A] [answered]
I added a plugin using slicer web cli task option and now I want to store work of each user by mapping it with it's unique ID or username, is there a way to extract such information?

**Comment by manthey** [comment]
If you have the `girderApiUrl` and `girderToken` passed to your client CLI, then you can use girder_client to ask for this information.  See the https://github.com/girder/slicer_cli_web/tree/458d616f2807b541b955a915cbb23e4575b8e8c3/example-girder-requests/girder_requests example. 

You'd do something like
```
gc = girder_client.GirderClient(apiUrl=args.girderApiUrl)
gc.setToken(args.girderToken)

userId = gc.get('token/current')['userId']
userLoginName = gc.get('user/' + userId)['login']
```

**Reply by lomshabhishek** [reply]
How can I get girderToken?

**Reply by manthey** [reply, marked as answer]
In your plugin, you can include parameters like https://github.com/girder/slicer_cli_web/blob/458d616f2807b541b955a915cbb23e4575b8e8c3/example-girder-requests/girder_requests/girder_requests.xml#L19-L36, and they will get passed in when the job is invoked.

**Reply by lomshabhishek** [reply]
@manthey You meant I just need to add the parameters in XML and then it would automatically get populated when the job is invoked?

**Reply by manthey** [reply]
Yes.

**Comment by lomshabhishek** [comment]
Yes got that however I want to know from where I can get this value
programmatically.

On Thu, Oct 5, 2023 at 5:51 PM David Manthey ***@***.***>
wrote:

> In your plugin, you can include parameters like
> https://github.com/girder/slicer_cli_web/blob/458d616f2807b541b955a915cbb23e4575b8e8c3/example-girder-requests/girder_requests/girder_requests.xml#L19-L36,
> and they will get passed in when the job is invoked.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DigitalSlideArchive/digital_slide_archive/discussions/297#discussioncomment-7197617>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBHZA53ECT7NS2TX5GEN3ULX52Q4HAVCNFSM6AAAAAA5SLPK6KVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TCOJXGYYTO>
> .
> You are receiving this because you authored the thread.Message ID:
> <DigitalSlideArchive/digital_slide_archive/repo-discussions/297/comments/7197617
> @github.com>
>

-- 


Confidentiality Notice:
The information transmitted in this email is 
intended only for the person(s) or entity to which it is addressed and may 
contain proprietary, confidential and/or privileged material. If you have 
received this email in error, please contact the sender by replying and 
delete this email so that it is not recoverable.
If you are not the 
intended recipient(s), any retention, review, disclosure, distribution, 
copying, printing, dissemination, or other use of, or the taking of any 
action in reliance upon, this information is strictly prohibited and 
without liability on our part. This email has been checked for any viral 
content, however we take no responsibility for the forwarding process of 
this email and recommend that this message and any attachment(s) are 
scanned by your system for any potential viral content.
